### DISCUSSION TITLE
When using `quarto use template` is it possible to not put the qmd file in a directory?

### DISCUSSION THREAD

**Discussion by dgkeyes** [quarto-dev/quarto-cli#8282] [Q&A]
### Description

I've created a Quarto extension and given people instructions to install it:

![image](https://github.com/quarto-dev/quarto-cli/assets/13971585/0c71da75-5a90-41f7-9b47-e87101e3372b)

It seems like the only way to use `quarto use template` is to have the qmd file that is created go into a separate directory. Is there a way to skip this and have it go in the root directory of the project users are working in?

**Comment by cderv** [comment]
No I think this is not possible for now. 

@dragonstyle is there a specific reason to require empty directory ?

**Reply by mcanouil** [reply]
FYI, it's not just about extensions. Extension behaviour is consistent with Quarto. Quarto HAS to create the directory itself (which is a bit annoying).
Quarto commands cannot take current directory name as name of the project to fill a directory already created.
For instance:
```sh
mkdir demo
cd demo
quarto create project website .
```

This create a project with "." everywhere and you have to give a name.
If you do give a directory name, it creates it.

**Reply by dragonstyle** [reply]
For templates, if the directory is empty, you will be prompted whether to use a subdirectory:

```bash
$ quarto use template quarto-journals/acm

Quarto templates may execute code when documents are rendered. If you do not
trust the authors of the template, we recommend that you do not install or
use the template.
 ? Do you trust the authors of this template (Y/n) › Yes
 ? Create a subdirectory for template? (y/n) ›
```

We always require a subdirectory if the current directory isn't empty as we will not know the files required to be copied by a template nor whether the template will work properly if some files are not copied (and we don't want to overwrite existing files without user confirmation).

**Reply by cderv** [reply]
Thanks for the explanation. I am thinking this could be worth adding to the doc at: https://quarto.org/docs/extensions/formats.html#format-templates 🤔

**Reply by mcanouil** [reply]
By the way, empty means really empty. Hidden files are accounted last time I've checked.

**Reply by dgkeyes** [reply]
Thanks, all, for the replies. Sounds like this isn't possible, which I totally understand.
